### PR TITLE
Fix replay buffer dtype

### PR DESCRIPTION
**Pull request by ahtsan**
Replay buffer should not have a default dtype, since each of the element in the replay buffer should have dtype same as the source, e.g. observation should have dtype same as env.observation_space. One example is DQN with pixel environment, we want the observation in replay buffer to be type `np.uint8`, same as the observation space.

### CODE REVIEW

File: tests/garage/replay_buffer/test_replay_buffer.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+from tests.fixtures.envs.dummy import DummyDiscreteEnv
+
+
+class TestReplayBuffer(unittest.TestCase):
+    def test_replay_buffer_dtype(self):
+        env = DummyDiscreteEnv()
+        obs = env.reset()
+        replay_buffer = SimpleReplayBuffer(
+            env_spec=env, size_in_transitions=100, time_horizon=1)
+        replay_buffer.add_transition(
```
Comment (ryanjulian) [inline tests/garage/replay_buffer/test_replay_buffer.py]: if this API operates on collections instead of single values, shouldn't it be `add_transitions`?

Comment (ahtsan) [reply tests/garage/replay_buffer/test_replay_buffer.py]: I think the API works specifically for `VecEnvExecutor`, where it adds a single transition for all of the n `vec_env`, resulting in a list of length n. If I have two `vec_env`, I will be calling `replay_buffer.add_transition(observation=[obs1, obs2], action=[act1, act2])`. But I think this assumption is not clear. We should add docstring that this replay buffer only works this way, rename it to something like `VecEnvReplayBuffer` and create another replay buffer which works with a single environment.
@CatherineSue Please correct me if I am wrong.

Comment (CatherineSue) [reply tests/garage/replay_buffer/test_replay_buffer.py]: No there is no such assumption.

Comment (CatherineSue) [reply tests/garage/replay_buffer/test_replay_buffer.py]: It adds a collection of transitions instead of a single transition

Comment (ahtsan) [reply tests/garage/replay_buffer/test_replay_buffer.py]: I see, we can also add more than one transition even when there is only one `vec_env`. I will rename the api and submit.

Comment (ryanjulian) [reply tests/garage/replay_buffer/test_replay_buffer.py]: if you would like to also have the `add_transition` API, it's trivial to define given `add_transitions` (just wrap the args in a list and call `add_transitions`)

**Pull request merged**